packages/next/src/server/config.ts

```diff
@@ -1382,7 +1411,58 @@ export default async function loadConfig(
       userConfig.htmlLimitedBots = userConfig.htmlLimitedBots.source
     }
 
-    onLoadUserConfig?.(Object.freeze(loadedConfig))
+    if (
+      debugPrerender &&
+      (phase === PHASE_PRODUCTION_BUILD || phase === PHASE_EXPORT)
+    ) {
+      userConfig.experimental ??= {}
+
+      setExperimentalFeatureForDebugPrerender(
+        userConfig.experimental,
+        'serverSourceMaps',
+        true,
+        reportExperimentalFeatures ? configuredExperimentalFeatures : undefined
+      )
+
+      if (process.env.TURBOPACK) {
```

Comment:
feel we can just use one call here and change the string by process.env.TURBOPACK to avoid writing the call twice.

`process.env.TURBOPACK ? 'turbopackMinify' : 'serverMinification'`